spark/src/main/scala/org/apache/spark/sql/comet/execution/shuffle/CometShuffleExchangeExec.scala

```diff
@@ -219,21 +271,146 @@ object CometShuffleExchangeExec extends ShimCometShuffleExchangeExec {
       newPartitioning: Partitioning,
       serializer: Serializer,
       writeMetrics: Map[String, SQLMetric]): ShuffleDependency[Int, InternalRow, InternalRow] = {
-    val sparkShuffleDep = ShuffleExchangeExec.prepareShuffleDependency(
-      rdd,
-      outputAttributes,
-      newPartitioning,
-      serializer,
-      writeMetrics)
+    val part: Partitioner = newPartitioning match {
```

Comment:
Is there a difference between this and calling `ShuffleExchangeExec.prepareShuffleDependency` directly?

Reply:
We call `ShuffleExchangeExec.prepareShuffleDependency` before.

But now we need to overwrite `needToCopyObjectsBeforeShuffle` which is a private method in Spark. So we need to copy the code of `ShuffleExchangeExec.prepareShuffleDependency` here.